### PR TITLE
RBF for F-71A quota tests

### PR DESCRIPTION
**Pull request by aditipande96**
Signed-off-by: Aditi Pande <aditi.pande@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by aditipande96** [comment]
Collection logs:
========= 2275 tests collected in 15.29s =======

Test logs:
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_40637
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41151

**Comment by aditipande96** [comment]
Test logs:
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41127
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41128
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41151
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41757
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_41758

File: libs/csm/rest/csm_rest_quota.py [hunk 1]
```diff
@@ -124,7 +122,7 @@ def verify_get_set_user_quota(self, uid: str, payload: dict, verify_response=Fal
                     if key in ("enabled","max_size","max_objects","check_on_raw"):
                         continue
                     if key == "max_size_kb":
-                        payload.update({"max_size_kb":get_response["max_size"]/1000})
+                        payload.update({"max_size_kb":round(get_response["max_size"]/1024)})
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_quota.py:125]: Can u once check this behaviour with RGW?
I think for RGW it is 1024

File: libs/csm/rest/csm_rest_quota.py [hunk 2]
```diff
@@ -134,58 +132,71 @@ def verify_get_set_user_quota(self, uid: str, payload: dict, verify_response=Fal
...
+    
+    # pylint: disable=too-many-arguments
+    def verify_max_objects(self, max_size: int, max_objects: int, akey: str, skey: str,
+                           bucket: str):
         """
         Verify put object of random size fails after exceeding max number of objects limit
         """
         self.log.info("Perform Put operation of small size and N object")
         small_size = math.floor(max_size / max_objects)
+        small_size = int((small_size/(1024*1024)*100))
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_quota.py:170]: What is 100?

File: tests/csm/rest/test_capacity_quota.py [hunk 3]
```diff
@@ -251,33 +252,32 @@ def test_40634(self):
...
-                      self.bucket, self.akey, self.skey)
-        bucket_created = s3_misc.create_bucket(self.bucket, self.akey, self.skey)
-        assert bucket_created, "Failed to create bucket"
         test_cfg = self.csm_conf["test_40634"]
-        quota_type = test_cfg["quota_type"]
         enabled = test_cfg["enabled"]
         max_size = test_cfg["max_size"]
         max_objects = test_cfg["max_objects"]
-        payload = self.csm_obj.iam_user_quota_payload(quota_type,enabled,max_size,max_objects)
+        payload = self.csm_obj.iam_user_quota_payload(enabled,max_size,max_objects)
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:259]: Perform autopep8

File: tests/csm/rest/test_capacity_quota.py [hunk 4]
```diff
@@ -738,18 +741,18 @@ def test_41746(self):
...
+        m_size = resp.json()["capacity"]["s3"]["users"][0]["used_rounded"]
 
-        assert_utils.assert_equals(self.user_id, uid, "id is not equal")
+        assert_utils.assert_equals(self.uid, uid, "id is not equal")
         assert_utils.assert_equals(total_objects, t_obj, "Number of objects not equal")
         assert_utils.assert_equals(total_objects, num_objects, "Number of objects not equal")
         assert_utils.assert_equal(total_size, t_size, "Total Size mismatch found")
-        assert_utils.assert_equal(total_size, data_size, "Total Size mismatch found")
-        assert_utils.assert_greater_equal(total_size, m_size, "Total Used Size mismatch found ")
+        assert_utils.assert_equal(total_size, data_size*1024, "Total Size mismatch found")
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:750]: what is the unit of  total_size? I think it is byte. Can u please confirm

File: tests/csm/rest/test_capacity_quota.py [hunk 5]
```diff
@@ -738,18 +741,18 @@ def test_41746(self):
...
+        assert_utils.assert_greater_equal(m_size, total_size, "Total Used Size mismatch found ")
         self.log.info("##### Test ended -  %s #####", test_case_name)
 
 
-    @pytest.mark.skip("Feature not ready")
+    #@pytest.mark.skip("Feature not ready")
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:755]: If not required please remove

File: libs/csm/rest/csm_rest_quota.py (hunk 1 above)
Comment (aditipande96) [reply libs/csm/rest/csm_rest_quota.py:125]: yes, I have used 1024 only

File: tests/csm/rest/test_capacity_quota.py (hunk 5 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:755]: okay

**Review by kachhwahadivya — APPROVED** [review]

**Comment by RajeshPratapSInghChouhan** [comment]
please fix codacy errors

**Review by RajeshPratapSInghChouhan — APPROVED** [review]

File: tests/csm/rest/test_capacity_quota.py (hunk 3 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:259]: okay

**Comment by aditipande96** [comment]
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_40635
PASSED tests/csm/rest/test_capacity_quota.py::TestCapacityQuota::test_40636

**Pull request merged**